lib/streamlit/web/server/server.py

```diff
@@ -112,6 +118,22 @@ def start_listening(app: tornado.web.Application) -> None:
         start_listening_tcp_socket(http_server)
 
 
+def _get_ssl_options(
+    cert_file: Optional[str], key_file: Optional[str]
+) -> Union[ssl.SSLContext, None]:
+    if bool(cert_file) != bool(key_file):
```

Comment:
@sfc-gh-kbregula  So this will just check for the presence of bytes right? What happens in case of invalid files. Do we fallback to http? Is that conveyed appropriately to the user? We may want to add such test cases as well.

Reply:
I think the `ssl_ctx.load_cert_chain(cert_file, key_file)` line below should throw an `SSLError` if given invalid input, which in this case I believe would bubble up to the user and cause streamlit to exit with a nonzero error code.

This behavior seems fine to me as long as the error thrown by the `ssl` lib is sufficiently user-friendly (otherwise I'd advocate for repackaging it into something a bit more readable), but we should probably add some tests that the expected error is thrown

Reply:
I added validations and test cases when files are missing or invalid.